File: 백준/Silver/11478. 서로 다른 부분 문자열의 개수/서로 다른 부분 문자열의 개수.py
import sys
input = sys.stdin.readline

def sol():
    res = set()
    s = input().strip()
    for i in range(len(s)):
        left = i
        for j in range(i+1, len(s)+1):
            res.add(s[i:j])
    print(len(res))

sol()

File: 백준/Silver/11478. 서로 다른 부분 문자열의 개수/서로 다른 부분 문자열의 개수.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.HashSet;

public class Main {
	
	private static int		len;
	private static String	input;

	private static HashSet<String>	set;
	
	public static void main(String[] args) throws IOException {
		init();
		sol();
	}
	
	private static void init() throws IOException {
		BufferedReader	br = new BufferedReader(new InputStreamReader(System.in));
		
		input = br.readLine().trim();
		len = input.length();
		set = new HashSet<>();
	}
	
	private static void sol() {
		for (int i = 0; i < len; i++) {
			for (int j = i; j < len; j++) {
				set.add(input.substring(i, j+1));
			}
		}
		System.out.println(set.size());
	}
	
}